### PR TITLE
Integrate facets with galleries

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

This ensures that the classifications and categories are correct when in the Image and Video galleries. This requires that the same additional filters (MIME types or musts) are sent to list the classifications function.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/api/search/SearchResourceImpl.java [hunk 1]
```diff
@@ -224,7 +227,18 @@ public FacetSearchBean searchFacets(
             null,
             owner,
             new DefaultSearch());
+    // Add MIME types
     search.setMimeTypes(mimeTypes);
+    // Add 'musts'
+    Optional.of(musts)
+        .filter(ms -> !ms.isEmpty())
+        .map(ms -> ms.toArray(new String[] {}))
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/api/search/SearchResourceImpl.java:235]: The recommended way to convert a list to array in Java 8 seems to be
 `ms.toArray(new String[0])`

**Review by PenghaiZhang — COMMENTED** [review]
Looks good to me! I have only two queries.

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/api/search/SearchResourceImpl.java (hunk 1 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/api/search/SearchResourceImpl.java:235]: Good plan.

File: Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx [hunk 2]
```diff
@@ -370,8 +373,25 @@ const SearchPage = ({ updateTemplate }: TemplateUpdateProps) => {
...
+          default:
+            throw new TypeError(
+              "Unexpected `displayMode` for determining classifications listing function"
+            );
+        }
+      });
+
       setSearchPageOptions(state.options);
-      Promise.all([doSearch(state.options), listClassifications(state.options)])
+      Promise.all([doSearch(state.options), getClassifications(state.options)])
```
[the remark this answers is not in the document]
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx:394]: Ah yes, I see the confusion now. Will add type to `getClassifications` - which is just a constant pointing to a function (not a HOF).

**Comment by edalex-ian** [comment]
All addressed thanks @PenghaiZhang . BTY.

**Review by PenghaiZhang — APPROVED** [review]

**Pull request merged**